hikari/api/special_endpoints.py

```diff
@@ -2293,3 +2296,107 @@ def add_text_input(
         ModalActionRowBuilder
             The modal action row builder to enable call chaining.
         """
+
+
+class PollBuilder(abc.ABC):
+    """Builder class for polls."""
+
+    __slots__: typing.Sequence[str] = ()
+
+    @property
+    @abc.abstractmethod
+    def question_text(self) -> str:
+        """The question text for the poll."""
+
+    @property
+    @abc.abstractmethod
+    def answers(self) -> typing.Sequence[PollAnswerBuilder]:
+        """The answers for the poll."""
+
+    @property
+    @abc.abstractmethod
+    def duration(self) -> undefined.UndefinedOr[int]:
+        """The duration of the poll."""
+
+    @property
+    @abc.abstractmethod
+    def allow_multiselect(self) -> bool:
+        """Whether a user can select multiple answers."""
+
+    @property
+    @abc.abstractmethod
+    def layout_type(self) -> undefined.UndefinedOr[polls.PollLayoutType]:
+        """The layout type for the poll."""
+
+    @abc.abstractmethod
+    def add_answer(self, answer: PollAnswerBuilder) -> Self:
+        """Add an answer to the poll.
```

Comment:
We should warn people not to use this or outright remove it.

I think `add_poll_answer` could be renamed to `add_answer` and replace this one.

If people want to pass the answer builders directly, then then can just pass them to the list of answers when creating the builder